Added width option

dpi0 opened the pull request:
The `width` option allows users to set the size of the terminal window to their preferences instead of the hardcoded 1/2 width of the window size.

I've tested this option locally and have respected the default width setting of 1/2.

Also I've updated the README.md to showcase this option.

dpi0 committed:
Added width option

dpi0 committed:
Updated README.md with width option

itmecho commented:
That's a good idea, thanks for the PR!

Do you think it would be a good idea to also support the number being more than 1 and just setting the width to that number? i.e. `width = 0.4` results in 40% of the screen but `width = 40` results in a fixed width of 40 columns?

We probably also want handling for if someone passes in a negative number to just use the default value.

dpi0 committed:
Fixed deprecated vim.api.nvim_buf_set_option usage

dpi0 committed:
Added height option

dpi0 committed:
Fixed bug: Vim:E95: Buffer with this name already exists

Removed nvim_buf_set_name causing this bug which sets the full file name
for a buffer, also reported in
https://github.com/neovim/neovim/issues/20531
